### DISCUSSION TITLE
Remove duplicate Cisco devices that are strictly license related

### DISCUSSION THREAD

**Discussion by Chiniquy** [netbox-community/devicetype-library#1412] [Ideas]
The Cisco device library can be confusing. Looking at just a small selection of the Cisco Catalyst 9300 series shows many different models:

![image](https://github.com/netbox-community/devicetype-library/assets/10709382/538e6945-fcdd-4f62-b901-2d1ab03f675d)

However, note that these models do not contain the "-E" or "-A" at the end of the model name. Only once we go down to the "Ordering Information" do we see these letters show up:

![image](https://github.com/netbox-community/devicetype-library/assets/10709382/fa210e55-7174-4214-a854-94da6b33a5d1)

This is because the "-E" or "-A" only refers to what license you are purchasing with the switch.
I wish to emphasize that the underlying equipment is the same, only the license is different, and that license can be changed after the fact. 

Because of this, I have removed all references to licenses in our own Netbox implementation, so instead of 
C9300L-48P-4G-E
or
C9300L-48P-4G-A
It just reads
C9300L-48P-4G

To me this makes sense, since the underlying model is in fact C9300L-48P-4G.

**Because of this I suggest that we remove any references to licenses from Device Types in the library, especially when it leads to duplicate entries for the same physical Device Type. This will simplify the library and keep the focus on modeling the actual devices, not whatever license was purchased at the time of sale.** 

If keeping track of licenses is important, then a custom field can be used (or some other function). 

What do you guys think?

**Comment by ryanmerolle** [comment]
I have no issue with this.  The contributions are community/crowd sourced, so we take what people submit for the most part with minor adjustments.

I can see some people complaining because they would track the licenses in the hostname, maybe a comment in each file would help to describe this.

If I run a show version, does the license variation appear in the version model number?

**Reply by mtinberg** [reply]
In my case I match up the device_type part_number with SNMP ENTITY-MIB and the license level _is_ included in the reported model number for catalyst devices, so the simplest thing for me was to have device type entries for each model and license level.  We also wanted to track the license level because it's important for sparing, if something needs IP-Base/IP-Advanced/Network-Advantage it can't be replaced with LAN-Base/Network-Essentials without potentially part of the config not applying.  Maybe with "Smart Licensing" you could transfer features from one to the other but we've traditionally bought fixed perpetual licenses baked in with LLW instead of SmartNet, so we end up tracking each as a separate device type, even if the underlying hardware is the same. 

I haven't seen this same behavior on the C9300 series though, but I don't have any C9300L just plain C9300 and C9300X variants eg.

```
ENTITY-MIB::entPhysicalDescr.4000 = STRING: C9300-48UN
ENTITY-MIB::entPhysicalModelName.4000 = STRING: C9300-48UN
```

as opposed to say a Cat3850

```
ENTITY-MIB::entPhysicalDescr.1000 = STRING: WS-C3850-48P-L
ENTITY-MIB::entPhysicalModelName.1000 = STRING: WS-C3850-48P-L
```


As long as the base device type is in the library, copy-pasta'ing and changing the name/description/part_number for each of the sub-types is not terribly onerous, they are created exactly once for the lifetime of the install as the physical properties of a specific model never change and don't need ongoing maintenance.

**Reply by danner26** [reply]
I agree, I don't see a problem with this unless a `show version` displays the model number with the license option. @mtinberg when you run a `show version` can you confirm or deny if the `-A` or `-E` is present?

**Reply by RustyRoyce** [reply]
![show_ver](https://github.com/netbox-community/devicetype-library/assets/95763752/71323f02-a98c-4208-b6e7-4d8fee67da15)
The model name is not suffixed with -E or -A when performing show version.

**Reply by mtinberg** [reply]
`show version` doesn't show the license level as part of the model on C9300/C9300X or WS-C3850/WS-C3650 family devices, it describes the license in a separate section, it's SNMP which makes the hard distinction, and not on the newer C9300/C9300X families (I don't have any active/online C9300L to test).  I'm OK with _not_ including the license as part of the part_number on the C9300/C9300L/C9300X family, as that's consistent with `show version` and `snmpwalk entPhysicalModelName` and I can deal with needing to manually create the variants on the old Catalyst family (not that we have any new models that need to be added in my install) based on the main description yaml. 

I would think it prudent to add a README.md to the device-types/Cisco directory to document why this is done this way to head off later questions, if that doesn't break anyone's scripting to have non-yaml files present.

**Reply by ryanmerolle** [reply]
PRs for the fix and doc updates are welcome

**Reply by DanSheps** [reply]
I just checked, WS-C3850-48F-L on our Cat 3850.  So, anything not 9300 will need to stay on the old style system with the proper model.

Also, @danner26, `show inventory` is better compared to `show version` for the model number discussion.  Here is what a `show inventory` gives:

```
NAME: "c38xx Stack", DESCR: "c38xx Stack"
PID: WS-C3850-48F-L    , VID: V06  , SN: FOC2022X0JL

NAME: "Switch 1", DESCR: "WS-C3850-48F-L"
PID: WS-C3850-48F-L    , VID: V06  , SN: FOC2022X0JL

NAME: "StackPort1/1", DESCR: "StackPort1/1"
PID: STACK-T1-50CM     , VID: V01  , SN: MOC2001ABBT

NAME: "StackPort1/2", DESCR: "StackPort1/2"
PID: STACK-T1-50CM     , VID: V01  , SN: LCC2013G607

NAME: "Switch 1 - Power Supply A", DESCR: "Switch 1 - Power Supply A"
PID: PWR-C1-1100WAC    , VID: V01  , SN: LIT19511CC3

NAME: "Switch 1 - Power Supply B", DESCR: "Switch 1 - Power Supply B"
PID: PWR-C1-1100WAC    , VID: V01  , SN: LIT202302UY

NAME: "Switch 1 FRU Uplink Module 1", DESCR: "2x1G 2x10G Uplink Module"
PID: C3850-NM-2-10G    , VID: V01  , SN: FOC19112G35
```

`show inventory` is what that SNMP mib is polling basically.  `show version` is just a quick aggregation of a bunch of other information from different areas and it might not be 100% consisitent across all devices as far as how things are reported.
